[CleanUp] Cleans Garnet Spec Implementation

eliasjpr opened the pull request:
Controller Specs
- Changes how Controller tests are performed it now requires less
boilerplate code

System Tests

- Removes unecessary modules and simplifies code a lot.
- No need to specify Capabilities
- Class change from GarnetSpec::System::Test to GarnetSpec::SystemTest

```crystal
require "./spec_helper"
include GarnetSpec::RequestHelper

describe "Request Specs" do
  describe "#get" do
    it "should return a HTTP::Client::Response" do
      get("/posts").should be_a HTTP::Client::Response
    end
  end
end
```

To enable Amber tests with garnet just have to overwrite the Garnet
handler in spec helper

```crystal
handler = Amber::Server.instance.handler
handler.prepare_pipelines

module GarnetSpec
  HANDLER = handler
end
```

eliasjpr committed:
[CleanUp] Cleans Garnet Spec Implementation

Controller Specs
- Changes how Controller tests are performed it now requires less
boilerplate code

System Tests

- Removes unecessary modules and simplifies code a lot.
- No need to specify Capabilities
- Class change from GarnetSpec::System::Test to GarnetSpec::SystemTest

```crystal
require "./spec_helper"
include GarnetSpec::RequestHelper

describe "Request Specs" do
  describe "#get" do
    it "should return a HTTP::Client::Response" do
      get("/posts").should be_a HTTP::Client::Response
    end
  end
end
```

To enable Amber tests with garnet just have to overwrite the Garnet
handler in spec helper

```crystal
handler = Amber::Server.instance.handler
handler.prepare_pipelines

module GarnetSpec
  HANDLER = handler
end
```

katafrakt commented on spec/controller/test_spec.cr:
```diff
@@ -1,77 +0,0 @@
-      headers = response["headers"]?
-      headers.should_not be_nil
-      headers.not_nil!.["Content-Type"].should eq ["application/json"]
-    end
```
Those specs are gone after changes. Is it possible to bring them back?

eliasjpr replied:
Yes you’re right :)

eliasjpr replied:
I have added these back @katafrakt

eliasjpr committed:
[Fixes] Gracefully Shuts down the GarnetSpec Server

Previously servers stay alive after a spec runs.

- Changes the way server shutdown is handled

Servers are shut down gracefully.

eliasjpr committed:
[Api] Adds Page method to System Test API

Having the page object available in tests is useful to perform custom
checks and provides a number of methods for controlling the navigation
of the page, such as visit, click, fill etc. The methods facilitates the
interaction with the current session.

eliasjpr commented:
@robacarp @drujensen @faustinoaq please review this PR and let me know if this will be merged. Thanks

eliasjpr commented on spec/spec_helper.cr:
```diff
@@ -1,29 +1,13 @@
-  # Not all server implementations will support every WebDriver feature.
-  # Therefore, the client and server should use JSON objects with the properties
-  # listed below when describing which features a session supports.
-  capabilities = {
```
There is no need to provide these options in spec helper since there are defaulted anyways.

eliasjpr commented:
Ping! @amberframework/contributors
